File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/faq/FaqCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.faq;

import java.util.List;
import ma.zs.alc.bean.core.faq.Faq;
import ma.zs.alc.dao.criteria.core.faq.FaqCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface FaqCollaboratorService extends  IService<Faq,FaqCriteria>  {

    List<Faq> findByFaqTypeId(Long id);
    int deleteByFaqTypeId(Long id);
    long countByFaqTypeId(Long id);




}

File: frontend/src/app/shared/service/collaborator/faq/FaqCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {FaqDto} from '../../../model/faq/Faq.model';
import {FaqCriteria} from '../../../criteria/faq/FaqCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";
import {Observable} from "rxjs";


@Injectable({
    providedIn: 'root'
})
export class FaqCollaboratorService extends AbstractService<FaqDto, FaqCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/faq/';
    }

    public constrcutDto(): FaqDto {
        return new FaqDto();
    }

    public constrcutCriteria(): FaqCriteria {
        return new FaqCriteria();
    }

    findAllGroupedByFaqType(): Observable<FaqDto[]> {
        return this.http.get<FaqDto[]>(`${this.API}grouped`);
    }
}
